tests/php/Helpers/ValidateSchema.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Helper used to validate schema differences.
+ *
+ * @package WooCommerce\Blocks\Tests
+ */
+
+namespace Automattic\WooCommerce\Blocks\Tests\Helpers;
+
+/**
+ * Validate schema.
+ */
+class ValidateSchema {
+	/**
+	 * The schema.
+	 *
+	 * @var array
+	 */
+	protected $schema = [];
+
+	/**
+	 * Constructor passed schema object.
+	 *
+	 * @param array $schema API schema representation.
+	 */
+	public function __construct( $schema ) {
+		$this->schema = $schema;
+	}
+
+	/**
+	 * Compare an object to the schema and return the diff.
+	 *
+	 * @param object $object Object to compare.
+	 * @return array
+	 */
+	public function get_diff_from_object( $object ) {
+		$schema_diff = array_diff(
+			array_keys( $this->schema['properties'] ),
+			array_keys( $object )
+		);
+
+		foreach ( $this->get_schema_with_nested_properties() as $property_name => $property_schema ) {
+			$property_value = current( $object[ $property_name ] );
+
+			if ( ! empty( $property_value ) ) {
+				$nested_diffs = array_diff(
+					array_keys( $property_schema['items']['properties'] ),
+					array_keys( $property_value )
+				);
```

Comment:
I don't think this catches additional nesting (which would explain how the image property mismatches got through the unit tests).  For example, if you have a response object (as an array) like this:

```php
$object = [
  'mainA' => [
    'propA' => 'a string',
    'propB' => [
      'subPropA' => 'some string',
      'subPropB' => 'some string',
    ],
  ],
];
```

And a schema like this (only including the requisite levels):

```php
$schema = [
  'mainA' => [
    'items' => [
      'properties' => [
        'propA' => $schema_args,
        'propB' => $schema_args,
      ],
    ],
  ],
];
```

Then you're not validating additional levels with this loop (which in my example would be the properties of the `propB` object`).

So essentially, you're only validating to the first level of `items.properties`